src/main/java/org/datadog/jmxfetch/App.java

```diff
@@ -213,27 +221,16 @@ public static int run(AppConfig config) {
 
     /** Attach a Shutdown Hook that will be called when SIGTERM is sent to JMXFetch. */
     private static void attachShutdownHook() {
-        class ShutdownHook {
-            public void attachShutDownHook() {
-                Runtime.getRuntime()
-                        .addShutdownHook(
-                                new Thread() {
-                                    @Override
-                                    public void run() {
-                                        LOGGER.info("JMXFetch is closing");
-                                        // Properly close log handlers
-                                        Enumeration<Appender> enume =
-                                                (Enumeration<Appender>) LOGGER.getAllAppenders();
-                                        while (enume.hasMoreElements()) {
-                                            Appender appender = enume.nextElement();
-                                            appender.close();
-                                        }
-                                    }
-                                });
+        Runtime.getRuntime().addShutdownHook(
+            new Thread() {
+                @Override
+                public void run() {
+                    LOGGER.info("JMXFetch is closing");
+                    // Properly close log handlers
+                    LogManager.shutdown();
```

Comment:
This was unnecessarily complicated (unless I'm missing something).  Hope you don't mind the cleanup.

Reply:
Definitely not! JMXFetch is known to have quite a bit of code that could use some cleanup, so this is definitely welcome.